streamer/src/sendmmsg.rs

```diff
@@ -135,14 +146,20 @@ where
     T: AsRef<[u8]>,
 {
     let size = packets.len();
-    #[allow(clippy::uninit_assumed_init)]
-    let iovec = std::mem::MaybeUninit::<iovec>::uninit();
-    let mut iovs = vec![unsafe { iovec.assume_init() }; size];
-    let mut addrs = vec![unsafe { std::mem::zeroed() }; size];
-    let mut hdrs = vec![unsafe { std::mem::zeroed() }; size];
+    let mut iovs = vec![MaybeUninit::<iovec>::uninit(); size];
+    let mut addrs = vec![MaybeUninit::<sockaddr_storage>::zeroed(); size];
+    let mut hdrs = vec![MaybeUninit::<mmsghdr>::uninit(); size];
     for ((pkt, dest), hdr, iov, addr) in izip!(packets, &mut hdrs, &mut iovs, &mut addrs) {
         mmsghdr_for_packet(pkt.as_ref(), dest.borrow(), iov, addr, hdr);
     }
+    // mmsghdr_for_packet() performs initialization so we can safely transmute
+    // the Vecs to their initialized counterparts
+    let _iovs = unsafe { std::mem::transmute::<Vec<MaybeUninit<iovec>>, Vec<iovec>>(iovs) };
```

Comment:
nit: import std::{mem, ptr} and use mem::transmute and ptr::write

Reply:
Done in https://github.com/anza-xyz/agave/pull/3325/commits/630c1dbb61e36704cd5095148b07b53ec48ef144

I did the same for another few instances for the sake of keeping use across the file consistent